Ft-Two-Factor-Authentication-#184359075

TresorRw opened the pull request:
**Adding Two-Factor Authentication**

Users that require a higher level of security (Sellers) should be able to turn on 2FA where authentication to the application also incudes a an extra factor i.e SMS or Authentications codes

Commit:
parent c7473be7c1b0f09f8c8570211333c8a2551fa25a
author Jean Damascene Uwingeneye <MacbookPro@Jeans-MacBook-Pro.local> 1675549448 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677243700 +0200

parent c7473be7c1b0f09f8c8570211333c8a2551fa25a
author Jean Damascene Uwingeneye <MacbookPro@Jeans-MacBook-Pro.local> 1675549448 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677243686 +0200

parent c7473be7c1b0f09f8c8570211333c8a2551fa25a
author Jean Damascene Uwingeneye <MacbookPro@Jeans-MacBook-Pro.local> 1675549448 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677243574 +0200

parent c7473be7c1b0f09f8c8570211333c8a2551fa25a
author Jean Damascene Uwingeneye <MacbookPro@Jeans-MacBook-Pro.local> 1675549448 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677243561 +0200

parent c7473be7c1b0f09f8c8570211333c8a2551fa25a
author Jean Damascene Uwingeneye <MacbookPro@Jeans-MacBook-Pro.local> 1675549448 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677243555 +0200

parent c7473be7c1b0f09f8c8570211333c8a2551fa25a
author Jean Damascene Uwingeneye <MacbookPro@Jeans-MacBook-Pro.local> 1675549448 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677243551 +0200

parent c7473be7c1b0f09f8c8570211333c8a2551fa25a
author Jean Damascene Uwingeneye <MacbookPro@Jeans-MacBook-Pro.local> 1675549448 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677243532 +0200

parent c7473be7c1b0f09f8c8570211333c8a2551fa25a
author Jean Damascene Uwingeneye <MacbookPro@Jeans-MacBook-Pro.local> 1675549448 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677243464 +0200

parent c7473be7c1b0f09f8c8570211333c8a2551fa25a
author Jean Damascene Uwingeneye <MacbookPro@Jeans-MacBook-Pro.local> 1675549448 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677243287 +0200

ch(coveralls) update jest config to add lcov.info in coverage

ch(coveralls) update github workflow

ch(coveralls) update github workflow

ch(coveralls) update github workflow

ch(coveralls) update github workflow to work on develop branch

ch(coveralls) coverage badge

ch(coveralls) coverage badge

update readme

feat(swaggerPackages): install packages

feat(swaggerSetup): resolve pull conflicts

feat(swaggerSamples): add few swagger samples

fix(eslint-typescript-any-error): fix eslint typescript any error authcontroller

fix: removed un wanted conflics

Added new route controller for sending and verifying the OTP

Added new routes for sending and verifying the OTP

Jest config file

deleted jest config file in main app diretory

Modified routes controller function for verifying the OTP

Used url params for getting user code and phone number

Re-structured the route for verifying the OTP

use of url params as verify/:phone/:code

Verification route controller function

Verification routes

Added package for twilio

fix: store jest files in root dir

test: logout functionality

config test: add coverage targets

update jest config

update jest config

fix: coveralls

fix: test env

Added the circleci configuration file

This file contains all about the configuration for the circleci

Added the CircleCI bade in the README file

This badge shows the status of the repo

Added spacing with badges

Changed badge

Lock

Separated app instance with database connecting

Removed console message

Commit:
parent c7473be7c1b0f09f8c8570211333c8a2551fa25a
author Jean Damascene Uwingeneye <MacbookPro@Jeans-MacBook-Pro.local> 1675549448 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677243700 +0200

parent c7473be7c1b0f09f8c8570211333c8a2551fa25a
author Jean Damascene Uwingeneye <MacbookPro@Jeans-MacBook-Pro.local> 1675549448 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677243686 +0200

parent c7473be7c1b0f09f8c8570211333c8a2551fa25a
author Jean Damascene Uwingeneye <MacbookPro@Jeans-MacBook-Pro.local> 1675549448 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677243574 +0200

parent c7473be7c1b0f09f8c8570211333c8a2551fa25a
author Jean Damascene Uwingeneye <MacbookPro@Jeans-MacBook-Pro.local> 1675549448 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677243561 +0200

parent c7473be7c1b0f09f8c8570211333c8a2551fa25a
author Jean Damascene Uwingeneye <MacbookPro@Jeans-MacBook-Pro.local> 1675549448 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677243555 +0200

parent c7473be7c1b0f09f8c8570211333c8a2551fa25a
author Jean Damascene Uwingeneye <MacbookPro@Jeans-MacBook-Pro.local> 1675549448 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677243551 +0200

parent c7473be7c1b0f09f8c8570211333c8a2551fa25a
author Jean Damascene Uwingeneye <MacbookPro@Jeans-MacBook-Pro.local> 1675549448 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677243532 +0200

parent c7473be7c1b0f09f8c8570211333c8a2551fa25a
author Jean Damascene Uwingeneye <MacbookPro@Jeans-MacBook-Pro.local> 1675549448 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677243464 +0200

parent c7473be7c1b0f09f8c8570211333c8a2551fa25a
author Jean Damascene Uwingeneye <MacbookPro@Jeans-MacBook-Pro.local> 1675549448 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677243287 +0200

ch(coveralls) update jest config to add lcov.info in coverage

ch(coveralls) update github workflow

ch(coveralls) update github workflow

ch(coveralls) update github workflow

ch(coveralls) update github workflow to work on develop branch

ch(coveralls) coverage badge

ch(coveralls) coverage badge

update readme

feat(swaggerPackages): install packages

feat(swaggerSetup): resolve pull conflicts

feat(swaggerSamples): add few swagger samples

fix(eslint-typescript-any-error): fix eslint typescript any error authcontroller

fix: removed un wanted conflics

Added new route controller for sending and verifying the OTP

Added new routes for sending and verifying the OTP

Jest config file

deleted jest config file in main app diretory

Modified routes controller function for verifying the OTP

Used url params for getting user code and phone number

Re-structured the route for verifying the OTP

use of url params as verify/:phone/:code

Verification route controller function

Verification routes

Added package for twilio

fix: store jest files in root dir

test: logout functionality

config test: add coverage targets

update jest config

update jest config

fix: coveralls

fix: test env

Added the circleci configuration file

This file contains all about the configuration for the circleci

Added the CircleCI bade in the README file

This badge shows the status of the repo

Added spacing with badges

Changed badge

Lock

Separated app instance with database connecting

Removed console message

TresorRw committed:
parent 594a16231d5240f29927cd7efd3f3f2ef7133927
author TresorRw <alaintresorcyusa683@gmail.com> 1675979701 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677242611 +0200

Added the circleci configuration file

This file contains all about the configuration for the circleci

Added the CircleCI bade in the README file

This badge shows the status of the repo

Added spacing with badges

Changed badge

Signup and Controls

Sign up and Token

add environment variables

fix: logout

fix: logout post route

FT: Update user role

Twilio version updated

setting up env

feat: codebase for login functionality

fixed login-functionality

resolve conficts and naming for "middlewares"

setting up env

feat: codebase for login functionality

fixed login-functionality

resolve conficts and naming for "middlewares"

Remove duplicate modules import

feat: created a token model

feat(reset-password):add resetpassword controller

fix: reset password token

fix: fixed the reset password

feat(reset-password): got 100% coverage

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Separated app instance with database connecting

Added return message when user does not provied phone number or OTP in verifying

This is a combination of 31 commits.

parent c693e66c3a170bc14456994f4710d4af53f1212c
author TresorRw <alaintresorcyusa683@gmail.com> 1676409350 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677094218 +0200

parent c693e66c3a170bc14456994f4710d4af53f1212c
author TresorRw <alaintresorcyusa683@gmail.com> 1676409350 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677094061 +0200

App instance

Testing the two-factor authentication

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

Added return message when user does not provied phone number or OTP in verifying

Replaced console with proper return messages

Added env var

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Added: @types/supertest package

Separated app instance with database connecting

Added return message when user does not provied phone number or OTP in verifying

App instance

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Added: @types/supertest package

Separated app instance with database connecting

Added return message when user does not provied phone number or OTP in verifying

App instance

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

Added env var

This is a combination of 3 commits.
This is the 1st commit message:

Added environment var for GA

This is the commit message #2:

Added env var into test.yml file

This is the commit message #3:

Added TWILIO CREDENTIALS env var into test.yml file

Removed console message

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Added return message when user does not provied phone number or OTP in verifying

This is a combination of 31 commits.

parent c693e66c3a170bc14456994f4710d4af53f1212c
author TresorRw <alaintresorcyusa683@gmail.com> 1676409350 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677094218 +0200

parent c693e66c3a170bc14456994f4710d4af53f1212c
author TresorRw <alaintresorcyusa683@gmail.com> 1676409350 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677094061 +0200

App instance

Testing the two-factor authentication

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

Added return message when user does not provied phone number or OTP in verifying

Replaced console with proper return messages

Added env var

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Added: @types/supertest package

Separated app instance with database connecting

Added return message when user does not provied phone number or OTP in verifying

App instance

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Added: @types/supertest package

Separated app instance with database connecting

Added return message when user does not provied phone number or OTP in verifying

App instance

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

Added env var

This is a combination of 3 commits.
This is the 1st commit message:

Added environment var for GA

This is the commit message #2:

Added env var into test.yml file

This is the commit message #3:

Added TWILIO CREDENTIALS env var into test.yml file

TresorRw committed:
parent 594a16231d5240f29927cd7efd3f3f2ef7133927
author TresorRw <alaintresorcyusa683@gmail.com> 1675979701 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677242611 +0200

Added the circleci configuration file

This file contains all about the configuration for the circleci

Added the CircleCI bade in the README file

This badge shows the status of the repo

Added spacing with badges

Changed badge

Signup and Controls

Sign up and Token

add environment variables

fix: logout

fix: logout post route

FT: Update user role

Twilio version updated

setting up env

feat: codebase for login functionality

fixed login-functionality

resolve conficts and naming for "middlewares"

setting up env

feat: codebase for login functionality

fixed login-functionality

resolve conficts and naming for "middlewares"

Remove duplicate modules import

feat: created a token model

feat(reset-password):add resetpassword controller

fix: reset password token

fix: fixed the reset password

feat(reset-password): got 100% coverage

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Separated app instance with database connecting

Added return message when user does not provied phone number or OTP in verifying

This is a combination of 31 commits.

parent c693e66c3a170bc14456994f4710d4af53f1212c
author TresorRw <alaintresorcyusa683@gmail.com> 1676409350 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677094218 +0200

parent c693e66c3a170bc14456994f4710d4af53f1212c
author TresorRw <alaintresorcyusa683@gmail.com> 1676409350 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677094061 +0200

App instance

Testing the two-factor authentication

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

Added return message when user does not provied phone number or OTP in verifying

Replaced console with proper return messages

Added env var

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Added: @types/supertest package

Separated app instance with database connecting

Added return message when user does not provied phone number or OTP in verifying

App instance

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Added: @types/supertest package

Separated app instance with database connecting

Added return message when user does not provied phone number or OTP in verifying

App instance

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

Added env var

This is a combination of 3 commits.
This is the 1st commit message:

Added environment var for GA

This is the commit message #2:

Added env var into test.yml file

This is the commit message #3:

Added TWILIO CREDENTIALS env var into test.yml file

Removed console message

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Added return message when user does not provied phone number or OTP in verifying

This is a combination of 31 commits.

parent c693e66c3a170bc14456994f4710d4af53f1212c
author TresorRw <alaintresorcyusa683@gmail.com> 1676409350 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677094218 +0200

parent c693e66c3a170bc14456994f4710d4af53f1212c
author TresorRw <alaintresorcyusa683@gmail.com> 1676409350 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677094061 +0200

App instance

Testing the two-factor authentication

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

Added return message when user does not provied phone number or OTP in verifying

Replaced console with proper return messages

Added env var

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Added: @types/supertest package

Separated app instance with database connecting

Added return message when user does not provied phone number or OTP in verifying

App instance

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Added: @types/supertest package

Separated app instance with database connecting

Added return message when user does not provied phone number or OTP in verifying

App instance

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

Added env var

This is a combination of 3 commits.
This is the 1st commit message:

Added environment var for GA

This is the commit message #2:

Added env var into test.yml file

This is the commit message #3:

Added TWILIO CREDENTIALS env var into test.yml file

TresorRw committed:
feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

TresorRw committed:
feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

TresorRw committed:
feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

TresorRw committed:
feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

TresorRw committed:
feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

TresorRw committed:
Removed consoles on send/verify OTP and used proper return for the error

TresorRw committed:
Removed consoles on send/verify OTP and used proper return for the error

TresorRw committed:
Added: @types/supertest package

TresorRw committed:
Separated app instance with database connecting

TresorRw committed:
Separated app instance with database connecting

TresorRw committed:
Added return message when user does not provied phone number or OTP in verifying

TresorRw committed:
Added return message when user does not provied phone number or OTP in verifying

TresorRw committed:
Added return message when user does not provied phone number or OTP in verifying

TresorRw committed:
Added return message when user does not provied phone number or OTP in verifying

TresorRw committed:
This is a combination of 31 commits.

parent c693e66c3a170bc14456994f4710d4af53f1212c
author TresorRw <alaintresorcyusa683@gmail.com> 1676409350 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677094218 +0200

parent c693e66c3a170bc14456994f4710d4af53f1212c
author TresorRw <alaintresorcyusa683@gmail.com> 1676409350 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677094061 +0200

App instance

Testing the two-factor authentication

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

Added return message when user does not provied phone number or OTP in verifying

Replaced console with proper return messages

Added env var

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Added: @types/supertest package

Separated app instance with database connecting

Added return message when user does not provied phone number or OTP in verifying

App instance

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Added: @types/supertest package

Separated app instance with database connecting

Added return message when user does not provied phone number or OTP in verifying

App instance

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

Added env var

This is a combination of 3 commits.
This is the 1st commit message:

Added environment var for GA

This is the commit message #2:

Added env var into test.yml file

This is the commit message #3:

Added TWILIO CREDENTIALS env var into test.yml file

TresorRw committed:
This is a combination of 31 commits.

parent c693e66c3a170bc14456994f4710d4af53f1212c
author TresorRw <alaintresorcyusa683@gmail.com> 1676409350 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677094218 +0200

parent c693e66c3a170bc14456994f4710d4af53f1212c
author TresorRw <alaintresorcyusa683@gmail.com> 1676409350 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677094061 +0200

App instance

Testing the two-factor authentication

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

Added return message when user does not provied phone number or OTP in verifying

Replaced console with proper return messages

Added env var

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Added: @types/supertest package

Separated app instance with database connecting

Added return message when user does not provied phone number or OTP in verifying

App instance

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Added: @types/supertest package

Separated app instance with database connecting

Added return message when user does not provied phone number or OTP in verifying

App instance

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

Added env var

This is a combination of 3 commits.
This is the 1st commit message:

Added environment var for GA

This is the commit message #2:

Added env var into test.yml file

This is the commit message #3:

Added TWILIO CREDENTIALS env var into test.yml file

TresorRw committed:
This is a combination of 31 commits.

parent c693e66c3a170bc14456994f4710d4af53f1212c
author TresorRw <alaintresorcyusa683@gmail.com> 1676409350 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677094218 +0200

parent c693e66c3a170bc14456994f4710d4af53f1212c
author TresorRw <alaintresorcyusa683@gmail.com> 1676409350 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677094061 +0200

App instance

Testing the two-factor authentication

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

Added return message when user does not provied phone number or OTP in verifying

Replaced console with proper return messages

Added env var

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Added: @types/supertest package

Separated app instance with database connecting

Added return message when user does not provied phone number or OTP in verifying

App instance

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Added: @types/supertest package

Separated app instance with database connecting

Added return message when user does not provied phone number or OTP in verifying

App instance

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

Added env var

This is a combination of 3 commits.
This is the 1st commit message:

Added environment var for GA

This is the commit message #2:

Added env var into test.yml file

This is the commit message #3:

Added TWILIO CREDENTIALS env var into test.yml file

TresorRw committed:
This is a combination of 31 commits.

parent c693e66c3a170bc14456994f4710d4af53f1212c
author TresorRw <alaintresorcyusa683@gmail.com> 1676409350 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677094218 +0200

parent c693e66c3a170bc14456994f4710d4af53f1212c
author TresorRw <alaintresorcyusa683@gmail.com> 1676409350 +0200
committer TresorRw <alaintresorcyusa683@gmail.com> 1677094061 +0200

App instance

Testing the two-factor authentication

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

Added return message when user does not provied phone number or OTP in verifying

Replaced console with proper return messages

Added env var

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Added: @types/supertest package

Separated app instance with database connecting

Added return message when user does not provied phone number or OTP in verifying

App instance

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

feat(verification): Added the swagger docs

- Added sending OTP to the user docs
- Added verifying the OTP provided by user docs

[Delivers #184359075]

feat(verification): changed response messages

- Changed the response on sending OTP
- Changed the response on verifying OTP

[Delivers #184359075]

Removed consoles on send/verify OTP and used proper return for the error

Added: @types/supertest package

Separated app instance with database connecting

Added return message when user does not provied phone number or OTP in verifying

App instance

Testing using the ENV VAR

Added new tests for incorrect phone or OTP

Replaced console with proper return messages

Added env var

This is a combination of 3 commits.
This is the 1st commit message:

Added environment var for GA

This is the commit message #2:

Added env var into test.yml file

This is the commit message #3:

Added TWILIO CREDENTIALS env var into test.yml file

TresorRw committed:
Lock

TresorRw committed:
Remove duplicate modules import

TresorRw committed:
Remove duplicate modules import

EmyRukundo commented on src/app.ts:
```diff
@@ -1,16 +1,11 @@
-
-app.use(authRoutes)
 
 console.log('Hello Team emma This backend API')
```
you can remove the console

EmyRukundo reviewed (changes requested):
Kindly squash the commits and good job in general

TresorRw committed:
Merge branch 'develop' of https://github.com/atlp-rwanda/e-comm-team-emma25-bn into develop

TresorRw committed:
Removed console message

TresorRw committed:
Merge branch 'ft-2fa-#184359075' of https://github.com/atlp-rwanda/e-comm-team-emma25-bn into ft-2fa-#184359075

TresorRw committed:
Merge branch 'ft-2fa-#184359075' of https://github.com/atlp-rwanda/e-comm-team-emma25-bn into ft-2fa-#184359075

TresorRw committed:
Merge branch 'ft-2fa-#184359075' of https://github.com/atlp-rwanda/e-comm-team-emma25-bn into ft-2fa-#184359075

TresorRw committed:
combination of 4 commits [squashed].

Resolved

Added bcrypt

Added logging: false in sequelizedb
Added --detect-openHandles flag in jest

Added --forceExit flag on test script

TresorRw committed:
combination of 4 commits [squashed].

Resolved

Added bcrypt

Added logging: false in sequelizedb
Added --detect-openHandles flag in jest

Added --forceExit flag on test script

TresorRw committed:
Merge branch 'ft-2fa-#184359075' of https://github.com/atlp-rwanda/e-comm-team-emma25-bn into ft-2fa-#184359075

TresorRw committed:
This is a combination of 3 commits

This is a combination of 3 commits that solves.

Solves timeout problem on long-running tests

Resolves: Conflicts after rebasing with develop branch

Resolved npm ERR! extraneous UNMET DEPENDENCY error

Increased timeout for long-running operations

Increased timeout on test

decreased timeout on test

TresorRw committed:
This is a combination of 3 commits that solves timeout on tests.

Solves timeout problem on long-running tests

Resolves: Conflicts after rebasing with develop branch

Resolved npm ERR! extraneous UNMET DEPENDENCY error

Increased timeout on tests

TresorRw committed:
Merge branch 'develop' into ft-2fa-#184359075

TresorRw committed:
Merge branch 'ft-2fa-#184359075' of https://github.com/atlp-rwanda/e-comm-team-emma25-bn into ft-2fa-#184359075

TresorRw committed:
Resdolve UNMET dependency

Resolved import problem

Re-tested and add the workflows secret
